### PR TITLE
feat(documentation): migrate docs page of subnavigation component

### PR DESCRIPTION
**Pull request by davidritter-dotcom**

### CODE REVIEW

File: packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { Meta, Source, Canvas } from '@storybook/blocks';
+import StylesPackageImport from '../../../shared/styles-package-import.mdx';
+import * as subnavigationStories from './subnavigation.stories';
+
+<Meta of={subnavigationStories} />
+
+# Subnavigation
+
+<p className="lead">This component elegantly organizes navigation links into intuitive horizontal tabs,
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx:9]: Was it written with ChatGPT? 😁
More seriously, we usually use action verbs for the lead. I would suggest something like: "Supplement the header navigation with a secondary entry-point, providing access to additional features in your application."

File: packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  </li>
+</ul>
+
+<Canvas sourceState="shown" of={subnavigationStories.Default} />
+
+<StylesPackageImport components={["subnavigation"]} />
+
+## Examples
+
+### Colored Background
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx:27]: Can you add a short explanation on this example and link to the [background utilities](https://preview-2574--swisspost-design-system-next.netlify.app/?path=/docs/utilities-background--docs)?

File: packages/documentation/src/stories/components/subnavigation/subnavigation.stories.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  },
+  render: render,
+  argTypes: {
+    coloredBackground: {
+      name: 'Colored Background',
+      control: {
+        type: 'boolean',
+      },
+    },
+    alternate: {
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/subnavigation/subnavigation.stories.ts:18]: The `.subnavigation-alternate` class only set a background color therefore it is no longer relevant: users can use the `.bg-light` class instead.

File: packages/documentation/src/stories/components/subnavigation/subnavigation.stories.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  return html`
+    <div
+      class="subnavigation${args.alternate
+        ? ' subnavigation-alternate'
+        : ''}${args.coloredBackground ? ' bg-petrol' : ''}"
+    >
+      <div class="container container-fluid-xs container-fluid-sm">
+        <ul class="subnavigation-list">
+          <li class="subnavigation-item">
+            <a href="/?path=/docs/components-subnavigation--docs" class="subnavigation-link active">
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/subnavigation/subnavigation.stories.ts:45]:
```suggestion
            <a href="#">
```

File: packages/documentation/src/stories/components/subnavigation/subnavigation.stories.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+              Navitem active
+              ${args.badge
+                ? html`
+                    <span class="badge bg-active rounded-pill">19</span>
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/subnavigation/subnavigation.stories.ts:49]: This classes aren't necessary: 
```suggestion
                    <span class="badge">19</span>
```

**Review by alizedebray — CHANGES_REQUESTED** [review]
Great work! Can you also: 
- [ ] Add the component to `preview.ts` so that it is correctly orders in the side nav
- [ ] Add the control table (it will probably only have a control for the bg color)
- [ ] Add a decorator that prevent page changes when clicking a nav (there is an example called "clickBlocker" in the card stories)

File: packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx (hunk 1 above)
Comment (davidritter-dotcom) [reply packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx:9]: Yes, this was a suggestion from ChatGPT :) Your suggestion sounds good, I will use this.

File: packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+To set the background of the component, simply use one of our [background utility](/?path=/docs/utilities-background--docs) classes.
+
+<Canvas of={subnavigationStories.ColoredBackground} />
+<div className="hide-col-default">
+  <Controls of={subnavigationStories.ColoredBackground} />
```
Comment (alizedebray) [inline packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx:36]: This is not needed anymore: 

```suggestion
```

**Review by alizedebray — APPROVED** [review]

**Comment by alizedebray** [comment]
👍

File: packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx (hunk 6 above)
Comment (davidritter-dotcom) [reply packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx:36]: Why would the controls to change the background in the background example not be needed?

Comment (davidritter-dotcom) [reply packages/documentation/src/stories/components/subnavigation/subnavigation.docs.mdx:36]: Why would the controls to change the background color in the background color example not be needed?

**Pull request merged**